pinot-broker/src/main/java/org/apache/pinot/broker/api/resources/PinotClientRequest.java

```diff
@@ -76,24 +85,28 @@ public String processQueryGet(
         requestJson.put(Request.DEBUG_OPTIONS, debugOptions);
       }
       BrokerResponse brokerResponse = requestHandler.handleRequest(requestJson, null, new RequestStatistics());
-      return brokerResponse.toJsonString();
+      asyncResponse.resume(brokerResponse.toJsonString());
     } catch (Exception e) {
       LOGGER.error("Caught exception while processing GET request", e);
       brokerMetrics.addMeteredGlobalValue(BrokerMeter.UNCAUGHT_GET_EXCEPTIONS, 1L);
-      throw new WebApplicationException(e, Response.Status.INTERNAL_SERVER_ERROR);
+      asyncResponse.resume(new WebApplicationException(e, Response.Status.INTERNAL_SERVER_ERROR));
     }
   }
 
   @POST
+  @ManagedAsync
   @Produces(MediaType.APPLICATION_JSON)
   @Path("query")
   @ApiOperation(value = "Querying pinot")
   @ApiResponses(value = {@ApiResponse(code = 200, message = "Query response"), @ApiResponse(code = 500, message = "Internal Server Error")})
-  public String processQueryPost(String query) {
+  public void processQueryPost(String query, @Suspended final AsyncResponse asyncResponse) {
     try {
       JsonNode requestJson = JsonUtils.stringToJsonNode(query);
+      if (requestJson.has("trace")) {
+        TimeUnit.SECONDS.sleep(15);
```

Comment:
What is this sleep for?

Reply:
ops, left over from my previous testing and it is supposed to be removed